src/converters/geometry_converters.ts

```diff
@@ -1,37 +1,28 @@
-import { BoundingBox, Geometry, Point, Polygon, Line } from '../classes';
+import { BoundingBox, Geometry, Polygon } from '../classes';
+import type { GeoJSONGeometry } from '../types';
 import { geometryToTurfBbox } from './turf/turf_converters';
 
 /**
  * Calculates the bounding box of a geometry
  * @param geometry
  */
-export function geometryToBoundingBox(geometry: Geometry): BoundingBox {
-  if (geometry instanceof BoundingBox) {
-    return geometry;
-  }
-
-  if (geometry instanceof Point) {
-    return new BoundingBox(geometry.coordinates.lon, geometry.coordinates.lat, geometry.coordinates.lon, geometry.coordinates.lat);
-  }
-
-  if (geometry instanceof Polygon || geometry instanceof Line) {
-    const bboxResult = geometryToTurfBbox(geometry);
-    return new BoundingBox(bboxResult[0], bboxResult[1], bboxResult[2], bboxResult[3]);
-  }
-
-  throw new Error('Geometry not supported');
+export function geometryToBoundingBox<G extends GeoJSONGeometry>(geometry: Geometry<G>): BoundingBox {
+  const bbox = geometryToTurfBbox(geometry);
+  return new BoundingBox(bbox);
 }
 
 /**
  * Calculates the bounding box of a feature and returns a polygon
  * @param boundingBox
  */
 export function boundingBoxToPolygon(boundingBox: BoundingBox): Polygon {
-  const points = new Array<Point>();
-  points.push(new Point(boundingBox.min.lon, boundingBox.min.lat));
-  points.push(new Point(boundingBox.max.lon, boundingBox.min.lat));
-  points.push(new Point(boundingBox.max.lon, boundingBox.max.lat));
-  points.push(new Point(boundingBox.min.lon, boundingBox.max.lat));
-  points.push(new Point(boundingBox.min.lon, boundingBox.min.lat));
-  return new Polygon([points]);
+  return new Polygon([
```

Comment:
Can't this be a deep clone of `boundingBox.geometry`?
https://stackoverflow.com/a/122704/11915280
Or do we think this should be more explicit? Also, the `BoundingBox` extends `Polygon`, do we really need this function?

Reply:
since bounding box inherits from polygon we actually don't need it

Reply:
Amazing ;)